app/controllers/locations_controller.rb

```diff
@@ -164,47 +174,35 @@ def valid_location_admin_district?(location_params)
-    if @location.update(deactivation_date: params[:location][:deactivation_date])
-      flash[:notice] = "#{@location.name || @location.postcode} has been deactivated"
+    if @location.update!(deactivation_date: @location.deactivation_date)
+      flash[:notice] = success_text
     end
     redirect_to scheme_location_path(@scheme, @location)
   end
 
-  def deactivation_date_errors
-    if params[:location][:deactivation_date].blank? && params[:location][:deactivation_date_type].blank?
-      @location.errors.add(:deactivation_date_type, message: I18n.t("validations.location.deactivation_date.not_selected"))
-    end
-
-    if params[:location][:deactivation_date_type] == "other"
-      day = params[:location]["deactivation_date(3i)"]
-      month = params[:location]["deactivation_date(2i)"]
-      year = params[:location]["deactivation_date(1i)"]
-
-      collection_start_date = FormHandler.instance.current_collection_start_date
-
-      if [day, month, year].any?(&:blank?)
-        { day:, month:, year: }.each do |period, value|
-          @location.errors.add(:deactivation_date, message: I18n.t("validations.location.deactivation_date.not_entered", period: period.to_s)) if value.blank?
-        end
-      elsif !Date.valid_date?(year.to_i, month.to_i, day.to_i)
-        @location.errors.add(:deactivation_date, message: I18n.t("validations.location.deactivation_date.invalid"))
-      elsif !Date.new(year.to_i, month.to_i, day.to_i).between?(collection_start_date, Date.new(2200, 1, 1))
-        @location.errors.add(:deactivation_date, message: I18n.t("validations.location.deactivation_date.out_of_range", date: collection_start_date.to_formatted_s(:govuk_date)))
-      end
+  def success_text
+    case @location.status
+    when :deactivated
+      "#{@location.name} has been deactivated"
+    when :deactivating_soon
+      "#{@location.name} will deactivate on #{@location.deactivation_date.to_formatted_s(:govuk_date)}"
     end
   end
 
   def deactivation_date
-    return if params[:location].blank?
-
-    collection_start_date = FormHandler.instance.current_collection_start_date
-    return collection_start_date if params[:location][:deactivation_date_type] == "default"
-    return params[:location][:deactivation_date] if params[:location][:deactivation_date_type].blank?
+    if params[:location].blank?
+      return
+    elsif params[:location][:deactivation_date_type] == "default"
+      return FormHandler.instance.current_collection_start_date
+    elsif params[:location][:deactivation_date].present?
+      return params[:location][:deactivation_date]
+    end
 
     day = params[:location]["deactivation_date(3i)"]
     month = params[:location]["deactivation_date(2i)"]
     year = params[:location]["deactivation_date(1i)"]
+    return nil if [day, month, year].any?(&:blank?)
 
-    Date.new(year.to_i, month.to_i, day.to_i)
+    Time.utc(year.to_i, month.to_i, day.to_i) if Date.valid_date?(year.to_i, month.to_i, day.to_i)
```

Comment:
```suggestion
    Time.local(year.to_i, month.to_i, day.to_i) if Date.valid_date?(year.to_i, month.to_i, day.to_i)
```

Always use local times. Rails does the conversion automatically.